service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala

```diff
@@ -257,6 +257,7 @@ case class RowCountCurator(protected val requestModelValidator: CuratorRequestMo
                 //remove all fields except primary key
                 //remove all sorts
                 val primaryKeyAliasFields = sourcePipeline.bestDimCandidates.map(dim => Field(dim.publicDim.primaryKeyByAlias, None, None)).toIndexedSeq
+                val primaryFilteredExpressions = sourcePipeline.requestModel.reportingRequest.filterExpressions.filter(filter => primaryKeyAliasFields.contains(Field(filter.field, None, None)))
```

Comment:
I don't think you want to remove all non-pk filters here.  Doesn't make sense given you want to have total rows reflect the number of rows with all filters applied.

Reply:
This is true.  I am considering other ways to approach this issue.

Reply:
Try seeing why the unexpected join is constructed in request model.